src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1252,6 +1252,18 @@
   (jdbc/do-commands
     "create index resource_params_cache_parameters_idx on resource_params_cache using gin (parameters)"))
 
+(defn fix-missing-edges-fk-constraint []
+  (when-not (sutils/constraint-exists? "certnames" "edges_certname_fkey")
+    (log/info "Cleaning up orphaned edges")
+    (jdbc/delete! :edges ["certname not in (select certname from certnames)"])
```

Comment:
I think inserting the valid data into a new table is less work than first deleting the invalid data and then rewriting the table.  

In both cases you rewrite the table but in the DELETE case you have to pay the price of all the deletes before you rewrite.